txnkv/txnsnapshot/scan.go

```diff
@@ -118,6 +118,12 @@ func (s *Scanner) Next() error {
 	if !s.valid {
 		return errors.New("scanner iterator is invalid")
 	}
+	if s.snapshot.interceptor != nil {
```

Comment:
I wonder what's the principle to place these interceptor settings.

Reply:
The principle is: trace the origin of the `ctx` associated with the inner gRPC request.

In this example, `ctx` is created on L117:
```go
bo := retry.NewBackofferWithVars(context.WithValue(context.Background(), retry.TxnStartKey, s.snapshot.version), scannerNextMaxBackoff, s.snapshot.vars)
```
So we immediately bind the Interceptor on L121:
```go
if s.snapshot.interceptor != nil {
	// User has called snapshot.SetInterceptor() to explicitly set an interceptor, we
	// need to bind it to ctx so that the internal client can perceive and execute
	// it before initiating an RPC request.
	bo.SetCtx(tikvrpc.SetInterceptorIntoCtx(bo.GetCtx(), s.snapshot.interceptor))
}
```